# In-Class Activity (July 14, 2026)

In [1]:
library(tidyverse)
library(moderndive)

── Attaching core tidyverse packages ─────────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.2.1     ✔ readr     2.2.0
✔ forcats   1.0.1     ✔ stringr   1.6.0
✔ ggplot2   4.0.3     ✔ tibble    3.3.1
✔ lubridate 1.9.5     ✔ tidyr     1.3.2
✔ purrr     1.2.2     
── Conflicts ───────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


In [2]:
wage_data <- read.table("data/wage.txt", header=TRUE)
head(wage_data)

,education,south,sex,experience,union,wage,age,race,occupation,sector,marr
,<int>,<int>,<int>,<int>,<int>,<dbl>,<int>,<int>,<int>,<int>,<int>
1,8,0,1,21,0,5.10,35,2,6,1,1
2,9,0,1,42,0,4.95,57,3,6,1,1
3,12,0,0,1,0,6.67,19,3,6,1,0
4,12,0,0,4,0,4.00,22,3,6,0,0
5,12,0,0,17,0,7.50,35,3,6,0,1
6,13,0,0,9,1,13.07,28,3,6,0,0


# Part A

In [3]:
# Linear model before adjusting for sex
wage_model <- lm(wage ~ education, data = wage_data)
get_regression_table(wage_model)

term,estimate,std_error,statistic,p_value,lower_ci,upper_ci
<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
intercept,-0.746,1.045,-0.714,0.476,-2.800,1.308
education,0.750,0.079,9.532,0.000,0.596,0.905


In [4]:
wage_model <- lm(wage ~ education + sex, data = wage_data)
get_regression_table(wage_model)

term,estimate,std_error,statistic,p_value,lower_ci,upper_ci
<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
intercept,0.218,1.036,0.210,0.834,-1.818,2.254
education,0.751,0.077,9.779,0.000,0.600,0.902
sex,-2.124,0.403,-5.273,0.000,-2.915,-1.333


"Adjusting for SEX" is the same thing as creating an additive model. Therefore, we fit the model using `wage ~ education + sex`. In other words, we are creating a linear model such that `y = beta_0 + beta_1*EDUCATION + beta_2*SEX + epsilon`.

This gives us the 95% confidence interval [0.600, 0.902]. We can see that this has changed from the confidence interval from the previously fitted linear model which was [0.596, 0.905]. However, the corresponding p-value for the education has not changed since it remains to be 0. The confidence intervals remain to be relatively similar before creating the additive model which indicates that there is not a strong relationship between education and sex. If there was a strong relationship between these two, then the confifdnce interval would have changed more significantly.

# Part B

The slope parameter beta_1 associated with education in this case adjusts for sex. Previously (in activity #2), we used education to predict wage which didn't account for sex. However we've now split apart the data for male and females so beta_1 now represents the relationship between education and wage when sex is held constant. What this means is that if we look at only the males we observe that an increase in education by 1 is associated with an increase by 0.751 in wage. Similarly, we would expect the same pattern for females. However, we now have a term for sex so just because the slopes are the same for education does not necessarily mean that the final predicted wage would be the same for a male or female with the same education level since there is now an additional slope term for sex.

# Part C

In [5]:
wage_model <- lm(wage ~ factor(occupation), data = wage_data)
get_regression_table(wage_model)

term,estimate,std_error,statistic,p_value,lower_ci,upper_ci
<chr>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>,<dbl>
intercept,12.704,0.630,20.154,0.000,11.466,13.942
factor(occupation): 2,-5.111,0.986,-5.183,0.000,-7.049,-3.174
factor(occupation): 3,-5.281,0.789,-6.693,0.000,-6.832,-3.731
factor(occupation): 4,-6.167,0.813,-7.587,0.000,-7.763,-4.570
factor(occupation): 5,-0.757,0.778,-0.972,0.331,-2.285,0.772
factor(occupation): 6,-4.278,0.733,-5.835,0.000,-5.718,-2.837


When fitting a linear model with occupation as the sole predictor variable and wage as the reponse variable, we need to account for the fact that occupation is a categorical variable rather than a continuous numerical variable. Using factor(), R will create dummy variables allowing us to separate out the different occupations and calculate slope for each one. In the table above, we can see that occupations 2, 3, 4, and 6 vary by a relatively significant amount compared to the reference level which in this case would be occupation 1. We can take a closer look using ANOVA.

In [6]:
anova(wage_model)

,Df,Sum Sq,Mean Sq,F value,Pr(>F)
,<int>,<dbl>,<dbl>,<dbl>,<dbl>
factor(occupation),5,2537.697,507.53945,23.22392,4.121742e-21
Residuals,528,11539.001,21.85417,NA,NA


The p-value in the table above indicates that wage for each occupation vary significantly from one another since the p value is 4.12*10^-21 which is essentially 0 and is therefore lower than 0.05. Therefore, there are significant wage differences among different occupations (for at least one of the occupations compared to the rest).